src/gerbil/runtime/mop.ss

```diff
@@ -219,14 +276,14 @@ namespace: #f
 ;; Given a struct super class (or false if none),
 ;; a list of super-classes (not including the class being created), and
 ;; a list of direct slots from the class being created,
-;; return the all-slots vector of slots (with #f in position 0) and the slot-table.
+;; return the slot-vector vector of slots (with #f in position 0) and the slot-table.
 ;; a table mapping symbol and keyword names to offset, and
 ;; : (OrFalse StructTypeDescriptor) (List TypeDescriptor) (List Symbol) \
 ;;   -> (Vector Symbol) (Table (Or Symbol Keyword) -> Fixnum)
 (def (compute-class-slots class-precedence-list direct-slots)
   (let* ((next-slot 1) ;; 0 is special slot for type-descriptor
          (slot-table (make-hash-table-eq))
-         (r-slots [#f])
+         (r-slots '(##type))
```

Comment:
that's a really bad name for a slot, let's leave it anonymous, or call it `__class` or something